src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ABSENT_INDEX = -1;
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int SHIFT = 1;
+    private static final int START = 0;
+    private Object[] elementData = new Object[DEFAULT_CAPACITY];
```

Comment:
Let's move initialization to the constructor.
```suggestion
    private Object[] values;
```